tests/integration/test_usage_tracking_integration.py

```
@@ -0,0 +1,119 @@
+    # 3. ASSERT
+    last_call = state_manager.session.last_call_usage
+    assert last_call["prompt_tokens"] == prompt_tokens
+    assert last_call["completion_tokens"] == completion_tokens
+    assert last_call["cost"] == pytest.approx(expected_cost)
+
+    session_total = state_manager.session.session_total_usage
+    assert session_total["cost"] == pytest.approx(expected_cost)
+
+
+@pytest.mark.asyncio
+async def test_session_total_accumulates_across_multiple_calls():
```

Pylint (3.3.7)

[refactor] Too many local variables (24/15)

(R0914)